src/ansys/rocky/core/launcher.py

```diff
@@ -82,15 +83,47 @@ def launch_rocky(
         else:
             raise RockyLaunchError(f"Port {server_port} is already in use.")
 
+    ansys_installations = get_available_ansys_installations()
+
     if rocky_exe is None:
-        awp_roots = [k for k in os.environ.keys() if k.startswith("AWP_ROOT")]
-        for awp_root in sorted(awp_roots, reverse=True):
-            rocky_exe = Path(os.environ[awp_root]) / "Rocky/bin/Rocky.exe"
-            if rocky_exe.is_file():
-                break
+        if rocky_version is None:
+            for installation in sorted(ansys_installations, reverse=True):
+                rocky_exe = (
+                    Path(ansys_installations[installation]) / "Rocky/bin/Rocky.exe"
+                )
+                if (
+                    rocky_exe.is_file()
+                    and installation >= MINIMUM_ANSYS_VERSION_SUPPORTED
+                ):
+                    break
+            else:
+                raise FileNotFoundError(
+                    "Rocky executable is not found."
+                )  # pragma: no cover
         else:
-            raise FileNotFoundError("Rocky executable is not found.")
-    else:
+            if isinstance(rocky_version, str):
```

Comment:
Why do you want to support `str`? This can lead the user to think it's possible to use `rocky_version=24.1`, which would be incorrect. Make it `int | None` only.